From a3046cf809529f4fdd03850e084a94fa72cf78e9 Mon Sep 17 00:00:00 2001
From: David Vrabel <david.vrabel@citrix.com>
Date: Wed, 10 Jun 2015 12:06:02 +0200
Subject: [PATCH] evtchn: profile event channel lock

The per-domain event channel lock may suffer from contention.  Add it to
the set of locks to be profiled when lock profiling is enabled.

Signed-off-by: David Vrabel <david.vrabel@citrix.com>
diff --git a/xen/common/event_channel.c b/xen/common/event_channel.c
index 5f92bc5..c0669ee 100644
--- a/xen/common/event_channel.c
+++ b/xen/common/event_channel.c
@@ -1260,7 +1260,7 @@ int evtchn_init(struct domain *d)
     if ( !d->evtchn )
         return -ENOMEM;
 
-    spin_lock_init(&d->event_lock);
+    spin_lock_init_prof(d, event_lock);
     if ( get_free_port(d) != 0 )
     {
         free_evtchn_bucket(d, d->evtchn);
